src/rect.rs

```diff
@@ -29,6 +29,24 @@ impl<T: fmt::Debug> fmt::Debug for Rect<T> {
     }
 }
 
+impl<T: fmt::Display> fmt::Display for Rect<T> {
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        // Why this won't compile? rustc will say that neither Point2D or
+        // Size2D has Debug implemented!
+        //write!(f, "Rect(Size {:?} at Origin {:?})", self.size, self.origin)
```

Comment:
Can you try `Size {} at Origin {}`? (or at least that with `self.size.to_string()` and likewise with `origin`)